### PR TITLE
feat(core-api): make it configurable whether to use estimates

### PR DESCRIPTION
**Pull request by vasild**
Introduce a new boolean config option totalCountIsEstimate and use
database estimates for the total number of rows if it is true (fast)
or use the precise COUNT(*) if the option is false (slow).

So, it is up to the node operator to configure their node for accuracy
vs speed. Add a new property in the response to indicate which one is
being used.

Resolves https://github.com/ArkEcosystem/core/issues/2676
Pagination error in api/blocks/{id}/transactions endpoint

A summary of what changes this PR introduces and why they were made.

## What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactor
- [ ] Performance
- [ ] Tests
- [ ] Build
- [ ] Documentation
- [ ] Code style update
- [ ] Continuous Integration
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Does this PR release a new version?

- [ ] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `develop` branch, _not_ the `master` branch
- [ ] All tests are passing
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: packages/core-database-postgres/src/defaults.ts [hunk 1]
```diff
@@ -11,4 +11,5 @@ export const defaults = {
         user: process.env.CORE_DB_USERNAME || process.env.CORE_TOKEN,
         password: process.env.CORE_DB_PASSWORD || "password",
     },
+    totalCountIsEstimate: true,
```
[the remark this answers is not in the document]
Comment (supaiku0) [reply packages/core-database-postgres/src/defaults.ts]: I think we should expose an env var for it e.g. `!process.env.CORE_API_NO_ESTIMATED_TOTAL_COUNT`

[the remark this answers is not in the document]
Comment (vasild) [reply packages/core-database-postgres/src/defaults.ts]: Done in 1a9dbeed0 and 15670bcb3.

File: packages/core-database-postgres/src/repositories/blocks.ts [hunk 2]
```diff
@@ -19,14 +20,18 @@ export class BlocksRepository extends Repository implements Database.IBlocksRepo
             } while (!first.operator && parameterList.length);
 
             if (first) {
-                selectQuery.where(this.query[this.propToColumnName(first.field)][first.operator](first.value));
+                for (const q of [ selectQuery, selectQueryCount ]) {
```
[the remark this answers is not in the document]
Comment (vasild) [reply packages/core-database-postgres/src/repositories/blocks.ts]: `for f in `git grep -l 'const q of'` ; do sed -E -i '' -e 's/const q of/const query of/' -e 's/([[:space:]]+)q\./\1query./' $f ; done`

[the remark this answers is not in the document]
Comment (vasild) [reply packages/core-database-postgres/src/repositories/blocks.ts]: 1cb028090

**Pull request merged**